### PR TITLE
Add CI github action

### PR DESCRIPTION
**Pull request by MalcolmMielle**
This PR adds a GitHub action to run `flutter analyze` on push and PR to main for validation. There is no test in the package yet, so the CI doesn't run any test ;).

I have added a similar CI in `adwaita` and also another action called `main.yml` to automatically publish the package to pub.dev anytime a new release is published. Let me know if you'd like to set up something similar here,

### CODE REVIEW

**Comment by prateekmedia** [comment]
Please create the release ci too, Also where will I found my secret credential json?

**Comment by MalcolmMielle** [comment]
The location of the pub-credential [depends on your OS](https://github.com/dart-lang/pub/issues/2999#issuecomment-908350917). If it's not there, it is in `pub-cache`, and you should `flutter pub logout` and `flutter pub login` again to have it at the right place.

**Comment by MalcolmMielle** [comment]
Release CI added.

File: README.md [hunk 1]
```diff
@@ -10,54 +13,58 @@ Unofficial implementation of Libadwaita Widgets in Flutter.
...
 
-### License
+## License
 
 `LGPL v3 / GNU LESSER GENERAL PUBLIC LICENSE v3`
 
 TLDR;
+
 - You are free to use [this](https://pub.dev/packages/libadwaita) package in whatever app you want,
 - If you improve the package then you should submit your patches / improvements to [this](https://github.com/prateekmedia/gtk-flutter) repository.
```
Comment (prateekmedia) [inline README.md:70]: Please also update the url if you are also Improving the README (:

Comment (MalcolmMielle) [reply README.md:70]: Good catch :)

File: .github/workflow/ci.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  test:
+    name: Test on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
```
Comment (prateekmedia) [inline .github/workflow/ci.yml:25]: If you are only running flutter analyze and only building the app on linux, then I don't think all platforms are needed as they all will report the same.

File: .github/workflow/lint.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+      ################################
+      # Run Linter against code base #
+      ################################
+      - name: Lint Code Base
+        uses: github/super-linter@v4
+        env:
+          VALIDATE_ALL_CODEBASE: false
+          DEFAULT_BRANCH: main
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (prateekmedia) [inline .github/workflow/lint.yml:53]: Why is github token needed for a lint checker?

File: .github/workflow/ci.yml (hunk 2 above)
Comment (MalcolmMielle) [reply .github/workflow/ci.yml:25]: I'll make a test on Adwaita to see if we could build the app for all platforms. If not, I'll remove the matrix test.

File: .github/workflow/lint.yml (hunk 3 above)
Comment (MalcolmMielle) [reply .github/workflow/lint.yml:53]: > If you pass the Environment variable GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} in your workflow, then the GitHub Super-Linter will mark the status of each individual linter run in the Checks section of a pull request. Without this you will only see the overall status of the full run. There is no need to set the GitHub Secret as it is automatically set by GitHub, it only needs to be passed to the action.

See [the linter action doc](https://github.com/marketplace/actions/super-linter#how-to-use)

**Review by prateekmedia — APPROVED** [review]
Thanks!
